akka-javasdk/src/main/java/akka/javasdk/UserException.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (C) 2021-2025 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk;
+
+public class UserException extends RuntimeException {
+
+  public UserException(String message) {
+    super(message);
+    setStackTrace(new StackTraceElement[0]);
```

Comment:
The idea was to have it defined in Scala and mixin `NoStackTrace`, but if we want to have it in java, then we should do the following instead:

```java
public class UserException extends RuntimeException  {
  public UserException(String message) {
    super(message);
  }

  @Override
  final public synchronized Throwable fillInStackTrace() {
    return this;
  }
}
```

This is the equivalent impl in java for `NoStackTrace`. 
The synchronized can proabably be removed.

Reply:
I also think that `setStackTrace` in the constructor won't work. 

The idea of `setStackTrace` is to remove the stack trace once we have one, but setting it to an empty array in the constructor won't prevent it to be written later.

Reply:
Ok, I'm wrong about the stack trace being filled later. The call happens in the constructor of `Throwable`. It calls `fillInStackTrace`, that calls a native method that will fill it. 

Then the `UserException` will reset it to empty. This will give the expected result, but partially. 

What we want is to not pay the price of filling the stack trace. That's why the Scala trait override `fillInStackTrace`. It doesn't get filled at all.

Reply:
yeah, overriding `fillInStackTrace` is nicer